src/server/pkg/sql/sql.go

```diff
@@ -0,0 +1,80 @@
+package sql
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+)
+
+// PGDumpReader parses a pgdump file into a header, rows, and a footer
+type PGDumpReader struct {
+	Header []byte
+	Footer []byte
+	rd     *bufio.Reader
+}
+
+// NewPGDumpReader creates a new PGDumpReader
+func NewPGDumpReader(r *bufio.Reader) *PGDumpReader {
+	return &PGDumpReader{
+		rd: r,
+	}
+}
+
+// ReadRows parses the pgdump file and populates the header and the footer
+// It returns EOF when done, and at that time both the Header and Footer will
+// be populated. Both header and footer are required. If either are missing, an
+// error is returned
+func (r *PGDumpReader) ReadRows(count int64) (rowsDump []byte, rowsRead int64, err error) {
+	if len(r.Header) == 0 {
+		err = r.readHeader()
+		if err != nil {
+			return nil, 0, err
+		}
+	}
```

Comment:
Does this reader have support for just reading a series of statements w/o reading a header and footer? That seems like something we should be able to support as well.

Reply:
What's the use case? For now I'm assuming that the input is a whole pgdump file that contains a single table's inserts. Right now we require both a header and footer, since that's what's needed to construct the table, etc. (Footer less so ... its just easier to require it for parsing reasons).

Reply:
Per offline discussion, we decided this was out of scope. However, we don't have a good answer today on how we want to handle incremental SQL data ingestion. [Issue here](https://github.com/pachyderm/pachyderm/issues/3164)